runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -664,4 +664,44 @@ public void shutdown() {
             streamingManager.shutdown();
         }
     }
+
+    /**
+     * Populate the ProtobufDescriptorTable using the RegistryTable.
+     * Enables backward compatibility in case of data migration.
+     */
+    public void syncProtobufDescriptorTable() {
+        log.info("Running syncProtobufDescriptorTable() ...");
+
+        int numRetries = 9;
+        while (true) {
+            if (TransactionalContext.isInTransaction()) {
+                throw new IllegalThreadStateException("openTable: Called on an existing transaction");
+            }
+            try {
+                this.runtime.getObjectsView().TXBuild().type(TransactionType.WRITE_AFTER_WRITE).build().begin();
+                registryTable.forEach((tableName, corfuRecord) ->
+                        corfuRecord.getPayload().getFileDescriptorsMap()
+                                .forEach((protoName, fileDescriptorProto) -> {
+                                    ProtobufFileName fileName = ProtobufFileName
+                                            .newBuilder().setFileName(protoName).build();
+                                    ProtobufFileDescriptor fileDescriptor = ProtobufFileDescriptor
+                                            .newBuilder().setFileDescriptor(fileDescriptorProto).build();
+                                    protobufDescriptorTable.putIfAbsent(fileName, new CorfuRecord<>(
+                                            fileDescriptor, corfuRecord.getMetadata()));
```

Comment:
It would also be useful to reset the registryTable's fileDescriptorsMap()'s values only to default instances here since they are no longer needed there. The keys in the registryTable's FileDesciptorMap should remain as is.
It is ok to do this as a follow up change too if you prefer. thanks